### DISCUSSION TITLE
Welcome to hell! A day in the life of a Maui developer.

### DISCUSSION THREAD

**Discussion by williambuchanan2** [dotnet/maui#15318] [General]
I thought it would be good to start a discussion on where all our time goes when working on Maui, and what our experiences are. I'll start...

Since our project is now so far behind - due to not being able to go more than 3 minutes without having to work around another Maui bug - we are working the weekends and evenings. Here is how I got on this weekend:

Saturday and Sunday - both days taken up trying to make a page scroll on iPhone. Page works perfect on Android (and did at one time work on iPhone) but now just drops off the bottom of the screen and doesn't scroll. Off course it turns out to be yet another well reported/unactioned bug. I lost 1 day working out if it was a bug or something I was doing wrong, then I spent the next day trying 1000 workarounds (during which time I kept being hindered by other bugs) before I got something to work.

Monday - an API call just suddenly stopped responding, seemingly for no reason. The server was responding but Maui was just hanging - no error message or any kind of response. The call stack just seemed to drop off. Calling the API in other places worked ok - so it's not the API itself. It's now 3pm and I just realised that the change which broke the API call was my removal of MaximumHeightRequest from a grid. Yes you read right - I removed MaximumHeightRequest from a grid, and that causes an API call to hang.
````
<Grid Margin="10" Padding="0" **MaximumHeightRequest="{Binding ScreenHeightAdjusted}"** HorizontalOptions="FillAndExpand" VerticalOptions="FillAndExpand">
````
Adding that height request back in makes the API work again. Strangely, it didn't occur to me until later in the day to revert that change and see if it fixed the API... Maybe some kind of error message would have been nice.

So, my output for 3 full days of work is that 1 page now scrolls (with a horrible workaround in place), and an API which stopped working now works again.

I've reached the point where I can't really be bothered to report bugs, because it is so pointless, however I did report one about DeviceInfo not working. It basically returns gibberish when you use it on an iPhone, so look out for that one!

**Comment by Xyncgas** [comment]
object oriented programming kinda do these things, I'm sorry you are suffering brother, I bet one day there's explainations like you are not writing program the way they want you to, but then the compiler should've caught this via type system which is the framework's oppertunity to ensure correctness, it might be without MaximumHeightRequest the Grid cause deadlock or stackoverflow it's poor design rather than your fault for using Grid inside this or that component for example

**Comment by BioTurboNick** [comment]
A network call caused the app to hang? I presume you checked to see that it wasn't a threading issue that just happens to be exposed or hidden?

Also curious if you read the documentation for DeviceInfo? Under Platform Differences:
https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/device/information?tabs=macios

**Reply by williambuchanan2** [reply]
No, it wasn't a one off caused by a network issue - it is consistently wrong. Yes read about the platform differences - these only relate to the Name, unless I am mistaken. In this case it reports all the wrong information. E.g. it thinks it is an iPhone 14 when it fact it is an iPhone 13.

**Reply by BioTurboNick** [reply]
> No, it wasn't a one off caused by a network issue - it is consistently wrong.

I don't mean a network issue. I mean a threading issue. As in a method is being called on the wrong thread. It's a common way for silent hangs to occur.

**Reply by Xyncgas** [reply]
could it have been prevented by the engine (rt) itself

**Reply by williambuchanan2** [reply]
Yeah I don't know. My view is these things should just work. No exceptions getting thrown, and get the same (wrong) result consistently.

**Comment by mjsb212** [comment]
I'm in the same boat & feel your pain, I've got the same "A day in the life" experience...come on MSFT your making money hand over fist with ChatGPT and your stock has exploded -- so throw some $$$$$ to the MAUI team and hire more people ASAP.

**Comment by AlanAnderson-BlueScope** [comment]
I spent over a week with a crash in Android release. The same code works on Windows, iOS and even Android in debug. After convincing myself it was my bug, I tried various logging approaches in an attempt to get log information from Android release. Analogous to the old days of printf debugging I added enough Console.WriteLines to the android device log that I figured out it was a binding problem with a collectionview using an observablecollection. After checking that all resources were defined, spinning around in my chair a few times. I found an article about binding issues on iOS in release (not Android). The solution was to add the XamlCompilation attribute into the partial class of the page. It worked. Of course, if you report it as an issue, they want you to provide a sample after you have spent a week determining that there is a difference with the XamlCompilation between debug and release apparently that only shows it's ugly head on Android. I will say - this is no worse than Xamarin.Forms in the early days. You spent more time working around issues than writing the app. But hey, they will start over and rewrite everything with breaking changes because they don't have a source base that they need to maintain. 



[XamlCompilation(XamlCompilationOptions.Skip)]

**Reply by Xyncgas** [reply]
it seems like a big business rules engine
I bet it would make you happy if you can file an issue and get it fixed in 3 days

**Reply by Xyncgas** [reply]
Maybe if you have a place out side of work where you can just post issues you run into in a special thread just for you and everyone do it for free to fix it for you out of fun and passion it would be nice too
